File: 2022-convert-1d-array-into-2d-array/2022-convert-1d-array-into-2d-array.py
class Solution(object):
    def construct2DArray(self, original, m, n):
        #check if array can be converted to 2D array
        if (len(original) != m*n):
            #if not, return empty array
            return []
        
        newArray = []
        #range(start, stop, step)
        for i in range (0, len(original), n):
            if original[i:i+n] != []:
                newArray.append(original[i:i+n])
                
        return newArray
        
        

File: 2022-convert-1d-array-into-2d-array/2022-convert-1d-array-into-2d-array.java
class Solution {
    public int[][] construct2DArray(int[] original, int m, int n) {
        
        int[][] arr = new int[m][n];
        
        if((m * n) < original.length || (m * n) > original.length) 
            return new int[0][0];
        else
            for(int i = 0; i < m; i++)
                for(int j = 0; j < n; j++)
                arr[i][j] = original[(n * i) + j]; 
       return arr;
    }
}
